source/slang/slang-emit-c-like.cpp

```diff
@@ -2367,6 +2370,10 @@ void CLikeSourceEmitter::defaultEmitInstExpr(IRInst* inst, const EmitOpInfo& inO
         m_writer->emit("GroupMemoryBarrierWithGroupSync()");
         break;
 
+    case kIROp_NonUniformResourceIndex:
+        emitOperand(inst->getOperand(0), getInfo(EmitOp::General)); // Directly emit NonUniformResourceIndex Operand0;
+        break;
+
```

Comment:
Shouldn't you emit with the keyword "NonUniformResourceIndex" like following?
```
m_writer->emit("NonUniformResourceIndex(");
emitOperand(inst->getOperand(0), getInfo(EmitOp::General));
m_writer->emit(")");

```
It seems like we don't have a test targeting HLSL for "NonUniformResourceIndex".
Can you add one and make sure we don't regress?

Reply:
this is just the base case, half expecting to fail, instead of silently working. hlsl, glsl has their own overloads of this. i'll add suitable tests.